### PR TITLE
Misc fix master

### PR DESCRIPTION
**Pull request by regit**
- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/

Describe changes:
- fix crash in payload printable
- fix memory leak
- add some variables to YAML

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/295
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/79

### CODE REVIEW

**Comment by inliniac** [comment]
Ticket for the ASN1 leak https://redmine.openinfosecfoundation.org/issues/2109

**Comment by regit** [comment]
Outch sorry to have missed the issue.

File: src/util-print.c [hunk 1]
```diff
@@ -227,6 +227,7 @@ void PrintStringsToBuffer(uint8_t *dst_buf, uint32_t *dst_buf_offset_ptr, uint32
                         src_buf[ch] == '\n' ||
                         src_buf[ch] == '\r') ? (uint8_t)src_buf[ch] : '.');
     }
+    dst_buf[dst_buf_size - 1] = 0;
```
Comment (inliniac) [inline src/util-print.c]: When/why is this needed? The PrintStringsToBuffer function is essentially a wrapper around snprintf call(s).

Comment (regit) [inline src/util-print.c]: ASAN was detecting an invalid buffer read (strlen used internally by jansson) so I did add the sanity check to be sure we don't get out of buffer

Comment (inliniac) [inline src/util-print.c]: Do you have a bt of a ASAN error for this? Would like to see if this issue might have a deeper cause.

(thought I already posted a comment like this, but can't find it)

Comment (regit) [inline src/util-print.c]: Just created this https://redmine.openinfosecfoundation.org/issues/2154

**Comment by regit** [comment]
Related redmine issue: https://redmine.openinfosecfoundation.org/issues/2154

**Comment by regit** [comment]
Resubmitting with added patch
